spec/models/manageiq/providers/hawkular/middleware_manager_spec.rb

```diff
@@ -198,4 +198,101 @@ def timeline_domain_expectations(status)
       end
     end
   end
+
+  describe 'connectivity' do
+    let(:ems) { FactoryGirl.create(:ems_hawkular) }
+    let(:endpoint) { FactoryGirl.create(:endpoint) }
+    let(:config) do
+      {
+        :username => 'abcdefghij',
+        :password => '1234567890',
+        :host     => 'awesomehawkular.org',
+        :port     => 8080,
+        :tenant   => 'hawkular',
+      }
+    end
+    context 'connect' do
+      before do
+        allow(ems).to receive(:hostname).and_return(config[:host])
+        allow(ems).to receive(:port).and_return(config[:port])
+        allow(ems).to receive(:default_endpoint).and_return(endpoint)
+        allow(ems).to receive(:authentication_userid).and_return(config[:username])
+        allow(ems).to receive(:authentication_password).and_return(config[:password])
+      end
+
+      it 'non-ssl' do
+        expect(::Hawkular::Client).to receive(:new) do |params|
+          expect(params).to include(
+            :entrypoint  => URI::HTTP.build(config.slice(:host, :port)).to_s,
+            :credentials => config.slice(:username, :password),
+            :options     => config.slice(:tenant).merge(:verify_ssl => 1, :ssl_cert_store => nil)
+          )
+        end
+        ems.connect
+      end
+      it 'ssl-without-validation' do
+        allow(endpoint).to receive(:security_protocol).and_return('ssl-without-validation')
+        expect(::Hawkular::Client).to receive(:new) do |params|
+          expect(params).to include(
+            :entrypoint  => URI::HTTPS.build(config.slice(:host, :port)).to_s,
+            :credentials => config.slice(:username, :password),
+            :options     => config.slice(:tenant).merge(:verify_ssl => 0, :ssl_cert_store => nil)
+          )
+        end
+        ems.connect
+      end
+      it 'ssl' do
+        allow(endpoint).to receive(:ssl_cert_store).and_return('stub SSL cert store')
+        allow(endpoint).to receive(:security_protocol).and_return('ssl')
+        expect(::Hawkular::Client).to receive(:new) do |params|
+          expect(params).to include(
+            :entrypoint  => URI::HTTPS.build(config.slice(:host, :port)).to_s,
+            :credentials => config.slice(:username, :password),
+            :options     => config.slice(:tenant).merge(:verify_ssl => 1, :ssl_cert_store => endpoint.ssl_cert_store)
+          )
+        end
+        ems.connect
+      end
+    end
+
+    context 'validation' do
+      shared_examples 'handles' do |method|
```

Comment:
Hmmm... I think we also need a positive test. i.e., a test were the connection succeeds ;)
...and also one test to check that no validation is done if you pass `false` in the last parameter to `raw_connect`.